src/app/components/table/table.ts

```diff
@@ -5835,6 +5877,11 @@ export class ColumnFilter implements AfterContentInit {
     hide() {
         if (this.hasFilterNotBeenApplied()) {
             this.clearFilter();
+            // apply the previous filter if the user has not clicked the apply button
+            if (this.appliedFilter !== null) {
+                this.dt.filters[<string>this.field] = this.appliedFilter;
+                this.dt._filter();
```

Comment:
In hide,  which can be called programmatically and possibly in other places where I have not "tried" to  identify) you don't want to clear all filters which updates the table and then apply the old filter to restore the table,  if the filters never changed.

I personally work with tables that have over 5000 rows that are not lazy loaded.  I actually  added a "close" button to the body of the advance filters i display and then call the hide() method